src/main/java/io/javalin/core/compression/Gzip.kt

```diff
@@ -0,0 +1,42 @@
+package io.javalin.core.compression
+
+import java.io.OutputStream
+import java.util.zip.GZIPOutputStream
+
+/**
+ * DynamicCompressionHandler uses this to perform Gzip compression
+ * @see DynamicCompressionHandler
+ *
+ * @param level Compression level. Higher yields better (but slower) compression. Range 0..9, default = 6
+ */
+class Gzip(val level: Int = 6) {
+
+    init {
+        require(level in 0..9) {
+            "Valid range for parameter level is 0 to 9"
+        }
+    }
+
+    /**
+     * @param out The target output stream
+     * @param data data to compress
+     */
+    fun write(out: OutputStream, data: ByteArray) {
+        GzipWrapper(out, level).use {
+            it.write(data)
+        }
+    }
+}
+
+/**
+ * Kotlin wrapper for Java's GZIPOutputStream. Required so we can set gzip compression level
+ */
+class GzipWrapper(out: OutputStream, level: Int = 6): GZIPOutputStream(out, true) {
+
+    init {
+        require(level in 0..9) {
+            "Valid range for parameter level is 0 to 9"
+        }
+        this.def.setLevel(level)
+    }
+}
```

Comment:
Seems a bit excessive to validate this again. Also, is this class still required? You could do this inside `write`.

Reply:
The repeated validation is there because it's technically a standalone class, so my thinking was to make it robust in case it's ever invoked from anywhere else. But I've done something better now: I marked it private :D 

But yes, unfortunately the wrapper class is still required.  The "def" field that needs to be accessed to set gzip compression level is protected and doesn't have a setter. Inheritance was the only way I could get to it.

Reply:
You could do

```kotlin
object : GZIPOutputStream(outputStream, true) {
    init {
        this.def.setLevel(level)
    }
}
```
But it's not required. I'm fine with the current approach, especially if it's a private class.

Reply:
Brilliant! I didn't know I could do this object magic. That's just me being new to Kotlin.

This is way cleaner than a separate class, definitely changing!